tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -430,7 +452,29 @@ internal static DynamicMethod CreateEndMethodDelegate(Type integrationType, Type
              */
 
             Log.Debug($"Creating EndMethod Dynamic Method for '{integrationType.FullName}' integration. [Target={targetType.FullName}, ReturnType={returnType.FullName}]");
-            MethodInfo onMethodEndMethodInfo = integrationType.GetMethod(EndMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
+            MethodInfo onMethodEndMethodInfo = null;
+
+            try
+            {
+                onMethodEndMethodInfo = integrationType.GetMethod(EndMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
+            }
+            catch (AmbiguousMatchException)
+            {
+                // If the type defines multiple OnMethodEnd methods to work with both void return types and non-void return types,
```

Comment:
nit - consider factorizing this code with the one above

Reply:
Good call, I refactored the common code paths in ac173c08b1b9c3c39876e5b55ae9edacc90bbf39